### ISSUE TITLE
Define and test audio↔Julia roughness alignment as a perceptual training objective

### ISSUE THREAD

**Issue by rhythmatician** [rhythmatician/FractalSync#105]
Part of #77

## Coordination update: morphology matrix (#140), 2026-09-07

#140 now owns the shared audio↔Julia morphology feature matrix and controlled cross-axis experiment. **This issue remains the detailed owner of H1, rough ↔ smooth**, including the existing-code audit, extractor/teacher comparison, and roughness-objective ablation. Do not create another roughness implementation in #140.

Extend the color-independence requirement below to **color and edge glow**. Canonical roughness is measured from the agreed pre-palette/pre-glow representation; palette contrast, brightness, glow width/intensity, zoom, or rendering artifacts must not satisfy a geometry objective. Use #140's fixed-view protocol and explicit resolution/threshold sensitivity tests rather than assuming invariance.

With the #139 Julia Probe Instrument deployed, `R_visual(t)` refers to the actual Julia parameter `c_J(t)`, which may differ from rider position `b(t)`. The older `c(t)` notation below denotes the rendered parameter for visual scoring. Probe cycling must not inflate roughness through temporal jitter or hide inappropriate excursions behind a matching time average.

Keep roughness distinct from H2 thickness/fullness, H3 structural order/entropy, H4 density, and presentation brightness. Test their correlation instead of collapsing them into one complexity score. Share blinded held-out comparisons and anti-cheating controls with #140.

As already established in the architecture coordination comment, #109 owns any causal audio roughness feature's final definition/normalization/versioning, and #108 owns its observation inclusion. This issue may establish an evaluator or training target without requiring a new runtime channel. #103 remains a possible offline teacher source.

Additional closure evidence:

- [ ] Roughness extraction is shared with #140, not duplicated.
- [ ] Palette/glow interventions cannot alter canonical geometric roughness; view/resolution sensitivity is documented.
- [ ] With #139 active, score `J(c_J)` and retain separate trajectory/temporal diagnostics.
- [ ] H1's held-out keep/revise/drop verdict is recorded in #140's cross-axis matrix.

## Motivation

FractalSync has a unusually natural cross-domain perceptual correspondence available to it:

- rock / distorted electric-guitar timbre can sound **rough, harsh, noisy, inharmonic, dense, or smooth**;
- Julia sets can look **smooth/round** or **jagged/filigreed/rough** depending on `c`.

We want the learned Player to discover useful motion through Mandelbrot parameter space while making the roughness of `J(c(t))` track the roughness of the music over time.

This should be a **learned perceptual relationship**, not a hard-coded rule such as `distorted guitar => move to c=X`.

The research question is whether we can define robust, time-aligned musical and visual roughness measures and use their agreement as a training objective that measurably improves eyes-on coherence.

## Important: this is not greenfield

The repository already contains remnants of this idea:

- `backend/docs/EMOTIONAL_COHERENCE.md` explicitly proposes matching jagged/smooth Julia appearance to timbre/roughness, with edge density / gradient energy as a first visual proxy and spectral/inharmonicity-style audio roughness as possible musical evidence.
- `backend/src/visual_metrics.py` describes itself as measuring qualities including roughness, but the current `LossVisualMetrics.compute_all_metrics()` implementation only emits `temporal_change`; there is no active roughness metric there today.
- `frontend/src/components/TrainingPanel.tsx` still contains a `distortion_roughness_loss` field/display, apparently as stale or legacy UI surface rather than proof that a current authoritative loss exists.

So the first step is an audit: determine what was previously implemented, what is stale, and what should be preserved versus deleted. Do not create a second competing roughness implementation because old names happen to remain in the tree.

## Relationship to the architecture

Keep the existing responsibility seam:

```text
causal musical analysis              Mandelbrot / Julia geometry
          │                                    │
          └────────── PlayerPolicy ─────────────┘
                         │
                    Controls / c(t)
                         │
                       J(c)
```

The musical-analysis module should not know Mandelbrot geometry, and the geometry module should not contain musical rules.

The **relationship between them is learned through the Player and its training objectives**.

If a deterministic visual-roughness quantity becomes part of shared runtime/training geometry or `PlayerObservation`, its authoritative definition belongs in `runtime-core` under ADR 0001. A Python differentiable surrogate is permitted only where autograd requires it and must be explicitly parity-pinned rather than becoming independent authority.

This ticket is research first; do not assume the final Player needs an explicit `roughness` runtime input/output if the loss alone is sufficient.

## Core question

Can we define two comparable time series

```text
R_music(t)   = perceived / machine-estimated musical roughness
R_visual(t)  = geometric/perceptual roughness of J(c(t))
```

such that rewarding their alignment causes the learned Player to make Julia morphology feel more musically appropriate, without encouraging flicker, noise, maximum-complexity collapse, or rendering-specific hacks?

## Stage 0 — audit existing code and history

Before implementing anything:

- trace all current and historical references to `roughness`, `distortion_roughness_loss`, jaggedness, edge density, visual-interest metrics, etc.;
- determine whether an old trainer ever computed a real roughness objective or whether only docs/UI remnants remain;
- identify any useful prior tests/experiments;
- identify stale duplicate-authority code that should eventually be removed rather than resurrected.

Record the result in the issue/research artifact.

## Stage 1 — define visual roughness independently of color

The metric should characterize **the Julia set / its geometry**, not the current palette, brightness, or shader.

Evaluate a small set of candidate metrics, starting with simple robust baselines:

1. **edge density / gradient energy** on a canonical grayscale iteration/distance representation;
2. **multiscale total variation / gradient energy** so the score distinguishes coarse smooth boundaries from fine filigree;
3. a **boundary-complexity / box-counting-like slope** if it adds useful information at acceptable cost;
4. another geometry-native quantity already available from the canonical Map/Julia machinery if it is better than rendering pixels.

Questions to answer:

- Is the metric stable across render resolution?
- Is it stable across zoom / framing, or can it be normalized to be?
- Is it independent of color mapping?
- Does it distinguish visually smooth vs visibly jagged Julia sets the way a human expects?
- Is it continuous enough in useful regions of `c`-space to provide trainable signal?
- Is it cheap enough to compute during training, or should it be precomputed as another map/mip field over `c`?

Prefer a geometry-derived or canonical grayscale representation over RGB screenshots, because the model must not learn to satisfy "roughness" by changing color contrast.

Produce a visual roughness map/sample plot over representative `c` regions and inspect it against actual Julia renders.

## Stage 2 — define musical roughness

Evaluate a few plausible musical-side signals rather than declaring one textbook metric correct.

Candidate families include:

- psychoacoustic roughness / sensory-dissonance estimators;
- spectral inharmonicity;
- spectral flatness/noisiness;
- high-frequency modulation / flux evidence;
- learned machine-listening descriptors or embeddings from #103.

Requirements:

- runtime candidate signals must be causal if they are eventually exposed to the Player;
- offline/noncausal teacher estimates are allowed for supervision only;
- do not equate loudness, brightness, distortion, or onset density with roughness by definition — measure their correlation and keep the concepts separable;
- test on more than one production style when data is available, including future live/rehearsal/band recordings rather than only Tool studio masters.

If #103 produces a credible roughness/tension/texture teacher, this ticket may consume that output instead of inventing a weaker hand-engineered audio metric.

## Stage 3 — alignment objective

Start with the simplest objective that tests the hypothesis.

A useful baseline may combine level agreement and trend agreement, e.g. conceptually:

```text
L_level  = distance(normalize(R_visual), normalize(R_music))
L_slope  = distance(ΔR_visual, ΔR_music)
L_rough  = w1 * L_level + w2 * L_slope
```

or a windowed correlation/rank-correlation objective if that proves more robust.

Do not use cross-entropy unless the problem is deliberately discretized into categorical roughness states; roughness is naturally continuous.

Research explicitly:

- global vs per-song normalization;
- causal smoothing time constant;
- whether a short perceptual lag is appropriate and how to avoid future leakage;
- correlation vs pointwise/regression loss;
- derivative/trend agreement;
- robustness to outliers and section-level distribution shifts.

Do not reward only correlation if a trivial high-variance/flickery trajectory can game it.

## Anti-cheating / anti-collapse constraints

A successful roughness objective must not let the Player "win" by:

- jittering `c` rapidly;
- maximizing visual detail everywhere;
- camping at one maximally rough region;
- manipulating color/contrast instead of Julia geometry;
- moving far outside the Mandelbrot set into sparse dust and calling that "rough";
- destroying existing smoothness / Momentum / Shore behavior.

Evaluate roughness jointly with the existing continuity, realm/Shore, and visual-interest constraints.

The desired behavior is:

> rough music tends to produce rougher Julia morphology; smoother music tends to produce smoother Julia morphology; transitions between them remain intentional and physically coherent in `c`-space.

## Ablation / proof of value

This objective earns its place only if it improves the actual visualizer.

Compare at minimum:

1. baseline Player/trainer without roughness alignment;
2. baseline + musical/visual roughness objective;
3. alternative candidate roughness definitions if Stage 1/2 leave more than one credible option.

Measure:

- held-out alignment between `R_music(t)` and `R_visual(t)`;
- lagged/zero-lag correlation as a diagnostic, not the sole objective;
- change/trend agreement;
- existing transition / continuity / topology metrics to detect regressions;
- eyes-on evaluation of whether rough passages actually look rougher in the intended morphological sense.

A lower roughness loss without better eyes-on behavior is not sufficient.

## Relationship to existing issues

- **#77** — parent Wayfinder map; this is part of the learned `music state -> Player -> c -> J(c)` contract-compliance route.
- **#81** — related loss-design work for realm/Shore/Transition objectives. Roughness is an additional perceptual objective, not a replacement for topology/crossing losses.
- **#103** — self-hosted machine-listening teachers may provide better musical roughness / tension / texture targets.
- **#104** — the causal PlayerPolicy / auxiliary-head contract determines how learned musical representations reach the main visual policy. This ticket should not invent a parallel model architecture.
- **ADR 0001** — any deterministic shared geometry used by training and runtime remains Rust-authoritative.

This ticket is intentionally orthogonal to #100 / CycleBank identity diagnostics. Periodic timing may eventually coordinate when roughness changes occur, but it is not required to define the roughness correspondence itself.

## Acceptance / resolution

Close this research ticket with:

- an audit of the existing/stale roughness artifacts in the repo;
- a chosen visual roughness definition, with evidence that it tracks human-visible Julia morphology and is not just color/brightness;
- a chosen musical roughness target or teacher source;
- a reproducible time-aligned roughness diagnostic over training songs;
- a concrete differentiable/optimizable training objective or an explicit verdict that this correspondence is not useful enough;
- anti-cheating tests / metrics;
- an ablation showing whether the objective improves the main visual task;
- eyes-on examples demonstrating rough↔rough and smooth↔smooth correspondence without unacceptable jitter or topology regressions;
- a clear authority decision for any shared deterministic visual metric (`runtime-core` if shared runtime/training semantics are required).

## Non-goals

- Do not hard-code particular `c` locations for distorted guitars or other instruments.
- Do not make roughness a binary "clean/distorted" classifier unless evidence strongly demands it.
- Do not couple the musical analyzer directly to Mandelbrot geometry.
- Do not use RGB/color contrast as the canonical visual roughness metric.
- Do not redesign CycleBank here.
- Do not add a large symbolic music-theory ontology.
- Do not assume the legacy `distortion_roughness_loss` UI name represents a valid current implementation.

**Comment by rhythmatician** [comment]
2026-09-01 architecture coordination update: keep this as a perceptual-objective experiment, but subordinate the musical-side runtime feature decision to #109 (`AudioActivation/1`). If a causal roughness/timbre descriptor proves useful as Player input, #109 owns its final definition/normalization/versioning. This ticket still owns the separate question of whether aligning `R_music(t)` with a geometry-derived `R_visual(t)` is a useful training objective. Do not create a parallel roughness input contract here.
